File: backend/app/schemas/recommendations.py
from pydantic import BaseModel
from typing import Dict, List, Optional


class ArtistsList(BaseModel):
    artistsList: List[str]


class TrackData(BaseModel):
    uri: str
    isGeminiRecommended: bool
    popularity: Optional[int] = None


class ArtistRecommendation(BaseModel):
    artistName: str
    tracks: Dict[str, TrackData]


class RecommendationsResponse(BaseModel):
    recommendations: List[ArtistRecommendation]

File: frontend/src/models/recommendations.ts
export interface ArtistsList {
  artistsList: string[];
}

export interface TrackData {
  uri: string;
  isAIRecommended: boolean;
  popularity: number | null;
  isSelected: boolean;
}

export interface ArtistRecommendation {
  artistName: string;
  tracks: Record<string, TrackData>;
}

export interface RecommendationsResponse {
  recommendations: ArtistRecommendation[];
}
